api/client/system/inspect.go

```
@@ -43,61 +42,91 @@ func NewInspectCommand(dockerCli *client.DockerCli) *cobra.Command {
+		return dockerCli.Client().NodeInspectWithRaw(ctx, ref)
+	}
+}
+
+func inspectService(ctx context.Context, dockerCli *client.DockerCli) inspect.GetRefFunc {
+	return func(ref string) (interface{}, []byte, error) {
+		return dockerCli.Client().ServiceInspectWithRaw(ctx, ref)
+	}
+}
+
+func inspectTasks(ctx context.Context, dockerCli *client.DockerCli) inspect.GetRefFunc {
+	return func(ref string) (interface{}, []byte, error) {
+		return dockerCli.Client().TaskInspectWithRaw(ctx, ref)
 	}
 }
 
-func isErrorNoSwarmMode(err error) bool {
-	return strings.Contains(err.Error(), "This node is not a swarm manager")
+func inspectVolume(ctx context.Context, dockerCli *client.DockerCli) inspect.GetRefFunc {
+	return func(ref string) (interface{}, []byte, error) {
+		return dockerCli.Client().VolumeInspectWithRaw(ctx, ref)
+	}
+}
+
+func inspectAll(ctx context.Context, dockerCli *client.DockerCli, getSize bool, typeConstraint string) inspect.GetRefFunc {
+	var inspectAutodetect = []struct {
+		ObjectType      string
+		IsSizeSupported bool
+		IgnoreFailures  bool
+		ObjectInspector func(string) (interface{}, []byte, error)
+	}{
+		{"container", true, false, inspectContainers(ctx, dockerCli, getSize)},
+		{"image", true, false, inspectImages(ctx, dockerCli)},
+		{"network", false, false, inspectNetwork(ctx, dockerCli)},
+		{"volume", false, false, inspectVolume(ctx, dockerCli)},
+		{"service", false, true, inspectService(ctx, dockerCli)},
+		{"task", false, true, inspectTasks(ctx, dockerCli)},
+		{"node", false, true, inspectNode(ctx, dockerCli)},
+	}
+
+	return func(ref string) (interface{}, []byte, error) {
+		for _, inspectData := range inspectAutodetect {
+			if typeConstraint != "" && inspectData.ObjectType != typeConstraint {
+				continue
+			}
+			v, raw, err := inspectData.ObjectInspector(ref)
+			if err != nil {
+				if typeConstraint == "" && (apiclient.IsErrNotFound(err) || inspectData.IgnoreFailures) {
+					continue
```

I don't think we want to ignore all errors, only "not a swarm manager" errors.